### ISSUE TITLE
See about adding a fully/semi automated contributors icon list

### ISSUE THREAD

**Issue by ang-zeyu** [MarkBind/markbind#1646]
So we don't miss anyone out from [here](https://markbind.org/about.html) : )

Something like this, or one of the more common "circular icon" ones
- https://github.com/all-contributors/all-contributors

Research if (I have no clue how this works)
- we can make it fully automated preferably
- if not, what kind of semi automation could we do?

**Comment by ang-zeyu** [comment]
Hi @tlylt, thanks for checking it out.

Have you come across any fuss free options to make this fully automated? (e.g. generate from commit history during CI)

If not I think we can go with the bot, seems really easy to use.

**Comment by tlylt** [comment]
> Hi @tlylt, thanks for checking it out.
> 
> Have you come across any fuss free options to make this fully automated? (e.g. generate from commit history during CI)
> 
> If not I think we can go with the bot, seems really easy to use.

Hi @ang-zeyu , may I know what kind of contributions recognition we are looking at?
`all-contributors` is more powerful and we get to assign the type of contributions to the person.
But, if we are mainly on displaying contributors who have committed to the codebase, I would say the `contributors-img` option I mentioned is quite fuss-free, so adding the following to our readme:
<a href="https://github.com/markbind/markbind/graphs/contributors">
  <img src="https://contrib.rocks/image?repo=markbind/markbind" />
</a>

Made with [contributors-img](https://contrib.rocks).

Please let me know your thoughts on it or I can try research a bit more on this:)

**Comment by ang-zeyu** [comment]
no objections on using all contributors, I think including the type of contributions looks nice too.

Was wondering more for workflow wise, whether we could do (and whether *its easy to do so* -- don't think we should be expending too much on this):
1. PR successfully merged
2. Github actions invokes some combination of these [commands](https://allcontributors.org/docs/en/cli/usage) + maybe more to auto detect new contributors and updates the contributors list 

As for the contributior detection, we can do it based on the commit author (not github api call which is too costly (unless I'm mistaken =P)).

For the contribution type, maybe something like adopting the [conventional commits spec](https://www.conventionalcommits.org/en/v1.0.0/) for the project to extract the `feat / fix / etc.` keywords as the type.
Side note: conventional commits can help with automated release changelog generation too which we don't have yet.

Again, this is might all be unnecessarily overcomplicated for a simple task like this though, so I'm really open to using the bot unless you manage to research up anything else 🙂

**Comment by tlylt** [comment]
I looked around a little bit to see if there is anyone working on similar ideas of automatically adding contributors via all-contributors.

TLDR:
Proposed actions:
1. Add all-contributors bot to the repo
2. Update the developer guide on 1. contributing section (prompt devs to self-comment and add themselves) 2. project management section (explain the steps to take and things to note)
3. (Optional?) Manually add the past developers using the bot

I don't think I have the rights to add, but I could help with 2,3 if we are going with this.

---
I found [this issue](https://github.com/avinashkranjan/Amazing-Python-Scripts/issues/256), which tries to make a GitHub Action to automatically call all-contributors bot after a PR is Merged. The issue was closed without resolution as the repo no longer uses all-contributors. The strategy that they employ now looks like a combination of:
- contributors-img
- manually highlight certain members' profile

In that issue, some devs explored ways to use GitHub action to achieve it, mainly via auto-commenting. But there are some downsides mentioned for the approach.

I then looked around in the all-contributors repo issues page to try my luck there. There are two relevant issues:
- all-contributors/all-contributors#358
- all-contributors/all-contributors#359

While they did not come up with a solution that meets our needs, I think their discussions are noteworthy:
>I have two concerns. For one, I do not believe that we can automatically determine the type of contribution.

>My second and more important concern is that I think appreciation should not be automated. It should take a human to look at a contribution and decide whether or not to appreciate the contribution, and how. If it doesn't take any effort, what is it even worth?

While following conventional commit spec might resolve the 1st, I am leaning towards agreeing with the second concern. 
Also if we find that rewriting the comment to prompt all contributor is troublesome, [Github-saved-replies](https://docs.github.com/en/enterprise-server@3.0/github/writing-on-github/working-with-saved-replies/creating-a-saved-reply) was mentioned as a way to help with it.

Lastly, I also searched to see how others use all-contributors in their repo and some dedicate an issue to track all adding of contributors. E.g [here](https://github.com/alan-turing-institute/AutSPACEs/issues/1).

Sidenote on the automated release changelog, have you considered [semantic release](https://github.com/semantic-release/semantic-release)? I have not used it before but I heard good things about it:)

**Comment by ang-zeyu** [comment]
> > My second and more important concern is that I think appreciation should not be automated. It should take a human to look at a contribution and decide whether or not to appreciate the contribution, and how. If it doesn't take any effort, what is it even worth?

Agreed, good point raised. Also worth mentioning that many repos dont automate this at all (require the contributor to add their name to a `.md` list on-PR), which seemed a little cumbersome to me. But I think the bot is a better compromise between the above concern and PR-convenience.

> I then looked around in the all-contributors repo issues page to try my luck there. There are two relevant issues:
>
> read.md all-contributors/all-contributors#358
> Update git-cz to the latest version 🚀 all-contributors/all-contributors#359

not really needed but are these the correct links? 👀 

> Proposed actions:

Let's go with this (including `3.`) 👍 

For `1.`, I've just performed step 1 mentioned [here](https://allcontributors.org/docs/en/bot/installation) here which might be sufficient for testing / implementing, let me know if you need anything else that can't be PR-ed in!

**Comment by ang-zeyu** [comment]
> Update the developer guide on 1. contributing section (prompt devs to self-comment and add themselves) 2. project management section (explain the steps to take and things to note)

PS: let's try to phrase this in a way that makes it optional for devs (maintainers will add it in if not), think we should try to make the pr process not-too-cumbersome for first timers.

**Comment by tlylt** [comment]
Opss the correct links are here: all-contributors/app#358, all-contributors/app#359

**Comment by tlylt** [comment]
Testing...
@all-contributors add @tlylt for code
